src/wgpu_engine.rs

```diff
@@ -131,6 +132,14 @@ enum TransientBuf<'a> {
     Gpu(&'a Buffer),
 }
 
+#[derive(Error, Debug)]
+pub enum WgpuRecordingError {
+    #[error("buffer for indirect dispatch not in map")]
+    IndirectDispatchBufferNotInMap,
+    #[error("buffer for download not in map")]
+    DownloadBufferNotInMap,
+}
+
```

Comment:
Similar comments to the first, this doesn't help a developer make adjustments they need. The error doesn't feel like it's necessary.

Reply:
Yeah I should dig deeper into the errors and try to get what they mean and also document that right and maybe also ask in zulip for explanations. Currently it's just `.ok_or(Err("".into()))?` rewritten as enum variant